src/lib/interrupt.c

```diff
@@ -87,34 +95,40 @@ static int irq_register_child(struct irq_desc *parent, int irq, int unmask,
 static void irq_unregister_child(struct irq_desc *parent, int irq)
 {
 	struct irq_desc *child;
-	struct list_item *clist;
 	struct list_item *tlist;
 	struct irq_cascade_desc *cascade = container_of(parent,
 						struct irq_cascade_desc, desc);
+	struct list_item *clist = &cascade->child[SOF_IRQ_BIT(irq)];
 
 	spin_lock(&cascade->lock);
```

Comment:
This isn't changed by this commit and not in this series at all. this spin-lock protects lists of IRQ children, I don't think those lists are ever modified from an IRQ context? If they are, we certainly should fix this, but this can be a separate patch.

Reply:
best to rebase the fix into this one otherwise we can break bisect (and make it confusing for any debugger/tester).